test/test_mpl.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <sparrow/mp_utils.hpp>
+
+#include <vector>
+
+namespace sparrow
+{
+
+    static constexpr auto some_types = mpl::typelist<int, float>{};
```

Comment:
I'll need to add some comments to organize these tests before removing the draft mode.